### PR TITLE
FEATURE: Display number of validation errors for each tab in inspector

### PR DESCRIPTION
**Pull request by markusguenther**
What I did
Completed issue #2377. The number of errors now gets displayed as a badge at the top right corner. This helps to show, why the "Apply" button is disabled when errors are in other tabs and thus not visible.

How I did it
The "validationError" object was already available in the redux store. I used it to summarize a count for each tab by testing how many of the field ids, whose values are faulty, are contained inside a tab group. Finally, I created a notification badge for the "Tabs" component, which is displayed if the count is higher than 0.

How to verify it
Delete the value of for instance a mandatory field and check if a badge with the number 1 will appear.

Resolves:  #2377

### CODE REVIEW

**Comment by markusguenther** [comment]
As replacement for https://github.com/neos/neos-ui/pull/2397

Was not able to change the PR anymore. So I created a new one and used @lorissikora changes.
Thanks for the Idea and basic work Loris.

Hope it's in your interest.

**Comment by markusguenther** [comment]
Thanks to @lorissikora for bringing that topic on the table. This is a really nice feature.
Would be so happy if you find time to give neos a second try ;)

**Comment by dimaip** [comment]
Could you please attach a screenshot @markusguenther

**Comment by markusguenther** [comment]
Bildschirmfoto 2019-11-27 um 14 17 41
Little Screenshot :)

File: packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js [hunk 1]
```diff
@@ -277,12 +281,21 @@ export default class Inspector extends PureComponent {
...
+                                }, 0) : 0;
+
                             return (
                                 <TabPanel
                                     key={$get('id', tab)}
                                     id={$get('id', tab)}
                                     icon={$get('icon', tab)}
                                     groups={$get('groups', tab)}
+                                    notifications={notifications}
+                                    title={Boolean(notifications) && <Badge className={style.tabs__notificationBadge} label={String(notifications)}/>}
```
Comment (dimaip) [inline packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js]: Could we maybe have a more informative label for notifications? E.g. "3 properties in this tab are failing validation". That would require I18n ofc.

**Comment by dimaip** [comment]
I like the looks @markusguenther! Left one suggestion.

File: packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js (hunk 1 above)
Comment (markusguenther) [reply packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js]: Sure will take care

**Comment by markusguenther** [comment]
@dimaip Did you mean something like that?
Bildschirmfoto 2019-11-27 um 15 16 21

**Comment by dimaip** [comment]
Yes, so such tooltip would be shown on hover

**Comment by markusguenther** [comment]
Bildschirmfoto 2019-11-27 um 16 29 25

Added a tool tip and found sadly an UI bug.

**Comment by Sebobo** [comment]
Bit long label. Especially when we translate it to German.

Maybe just `General - 2 validation issues`?

**Comment by dimaip** [comment]
Typography nazi: also it should be a endash, not a minus sign :)

**Comment by markusguenther** [comment]
Changed the labels and added the special dash ;)
The plural groups are obviously not supported in the UI.

Try to create an PR for that and as the handler takes the first one it works for german and danish.
Just in english it is bit weird to have 2 issue and not issues.

**Comment by bwaidelich** [comment]
I really like this, looking at the screenshots you provided. But I failed to test this locally..
What I did:
```
git checkout -b markusguenther-lorissikora-2.x master
git pull https://github.com/markusguenther/neos-ui.git lorissikora-2.x
make setup
make build
```

(and set `frontendDevelopmentMode: true` ofc)

but no badges for me..

**Comment by markusguenther** [comment]
Thanks @bwaidelich ... bit late today (came back from fire brigade) but will check that tomorrow.

**Comment by markusguenther** [comment]
Bildschirmfoto 2019-11-30 um 13 34 14

Tested this on a fresh installation and had the same issue.
Reason was the error detection. So I now extracted that and now it works  in different setups.

@bwaidelich hope it now also works for you :)

**Comment by jonnitto** [comment]
thank you @markusguenther! This one is a very very important feature! Thank you for taking care! ❤️

File: packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js [hunk 2]
```diff
@@ -219,10 +222,24 @@ export default class Inspector extends PureComponent {
         this.setState({toggledPanels: newState});
     };
 
+    getAmountOfValidationErrors = (tab, validationErrors) => {
+        let errors = 0;
```
Comment (dimaip) [inline packages/neos-ui/src/Containers/RightSideBar/Inspector/index.js]: Btw, this could also be a `reduce` (no need to change anything now, just saying)

**Pull request merged**